guides/common/modules/proc_installing-the-dns-infoblox-module.adoc

```diff
@@ -1,8 +1,9 @@
 [id="Installing_the_DNS_Infoblox_Module_{context}"]
 = Installing the DNS Infoblox module
 
-Use this procedure to install the DNS Infoblox module on {SmartProxy}.
+ifeval::["{context}" != "{smart-proxy-context}"]
```

Comment:
It may not be the right way, but my reasoning is that the procedure is now included in both Provisioning Hosts (as it previously was) and in Installing Proxy, which is new. The xref only works in the Provisioning Hosts guide.

Perhaps the proper way is to create a separate file just for the procedure. That also allows using a consistent header as for the other DNS providers and a little introduction that's relevant.

Thoughts on how to move forward?

Reply:
Users can follow the first procedure (Installing the DHCP Infoblox module) and then the other (Installing the DNS Infoblox module), right? It will give the same result as "combining" both, right? If that's the case, I'd just drop the hint about combining the procedures.

This would be in line with the recommended approach in RH d/s to avoid documenting multiple ways of accomplishing the same thing. In other words, we're encouraged to pick one approach (the easiest one, the least error-prone one, the most user-friendly one, the 'best practices' one, ...) and avoid mentioning other approaches. Also, instructions to "combine" procedures might cause unnecessary confusion on the users' side ("Does 'combine' mean following one procedure and then the other? Or running all the installer options with one installer command? And what about the rest of the procedure steps, in what order should I follow them?").

Reply:
> Users can follow the first procedure (Installing the DHCP Infoblox module) and then the other (Installing the DNS Infoblox module), right? It will give the same result as "combining" both, right? If that's the case, I'd just drop the hint about combining the procedures.

Correct, both are independent. Logically, you could say it's A + B == B + A == AB.

> This would be in line with the recommended approach in RH d/s to avoid documenting multiple ways of accomplishing the same thing. In other words, we're encouraged to pick one approach (the easiest one, the least error-prone one, the most user-friendly one, the 'best practices' one, ...) and avoid mentioning other approaches. Also, instructions to "combine" procedures might cause unnecessary confusion on the users' side ("Does 'combine' mean following one procedure and then the other? Or running all the installer options with one installer command? And what about the rest of the procedure steps, in what order should I follow them?").

I agree, but there are many counter examples in our current documentation.

You could argue this PR is a step in untangling that mess. For example, documenting the nsupdate provider means we could pull the DNS part out of https://docs.theforeman.org/nightly/Installing_Server/index-katello.html#configuring-dns-dhcp-and-tftp_foreman. If we then do the same with DHCP then you remain with just TFTP installation. That's my ultimate goal with https://github.com/theforeman/foreman-documentation/issues/2957.

Thinking more about this, perhaps we need a generic section that talks about combining installer arguments **if** we want to encourage users to do so. There are good reasons, most come down to speed: the installer has a certain overhead so running it just once with more arguments is faster. Plus, if you install a DNS module you will restart services, refresh features. Then if you install a DHCP module, you do the same. If it's in a single run, you only need to restart and refresh once.

Reply:
> > This would be in line with the recommended approach in RH d/s to avoid documenting multiple ways of accomplishing the same thing. In other words, we're encouraged to pick one approach (the easiest one, the least error-prone one, the most user-friendly one, the 'best practices' one, ...) and avoid mentioning other approaches. Also, instructions to "combine" procedures might cause unnecessary confusion on the users' side ("Does 'combine' mean following one procedure and then the other? Or running all the installer options with one installer command? And what about the rest of the procedure steps, in what order should I follow them?").
> 
> I agree, but there are many counter examples in our current documentation.

Very true :)

> You could argue this PR is a step in untangling that mess. For example, documenting the nsupdate provider means we could pull the DNS part out of https://docs.theforeman.org/nightly/Installing_Server/index-katello.html#configuring-dns-dhcp-and-tftp_foreman. If we then do the same with DHCP then you remain with just TFTP installation. That's my ultimate goal with #2957.

I think it's the current messy-ish situation that prevents me from proposing anything useful when it comes to structuring this content. In https://github.com/theforeman/foreman-documentation/issues/2957, I proposed starting by defining the assemblies first. That's my preferred approach because reasonably defined assemblies can then be moved around guides with relative ease.

> 
> Thinking more about this, perhaps we need a generic section that talks about combining installer arguments **if** we want to encourage users to do so. There are good reasons, most come down to speed: the installer has a certain overhead so running it just once with more arguments is faster. Plus, if you install a DNS module you will restart services, refresh features. Then if you install a DHCP module, you do the same. If it's in a single run, you only need to restart and refresh once.

That's sounds like material for the installer man page rather than product guides (IMHO). I guess if this behavior was documented in the man page, it would be possible to reference it from the procedure modules we're discussing (from Additional resources). And _then_ you could drop the hint about combining the procedures..? But it's hard for me to say at this point.

Reply:
> That sounds like material for the installer man page rather than product guides

That sounds reasonable. But we could also mention that combining the installer args is faster when we first mention the installer in the Installing docs.